test/quantization/pt2e/test_numeric_debugger.py

```diff
@@ -188,3 +189,53 @@ def test_extract_results_from_loggers(self):
         for node_summary in comparison_results.values():
             if len(node_summary.results) > 0:
                 self.assertGreaterEqual(node_summary.results[0].sqnr, 35)
+
+    def test_export_retains_debug_handles(self):
+        class ConvAndRelu(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.conv = torch.nn.Sequential(
+                    torch.nn.Conv2d(
+                        in_channels=8,
+                        out_channels=8,
+                        kernel_size=1,
+                    ),
+                    torch.nn.ReLU(),
+                )
+
+            def forward(self, x):
+                return self.conv(x)
+
+        m = ConvAndRelu()
+        example_inputs = (torch.randn(1, 8, 4, 4),)
+        ref_exported = export_trace._export(m, example_inputs, pre_dispatch=True)
+        m = ref_exported.module()
+        generate_numeric_debug_handle(m)
+        ref_handle_to_nodes = defaultdict(list)
+        for n in m.graph.nodes:
+            if (
+                CUSTOM_KEY in n.meta
+                and NUMERIC_DEBUG_HANDLE_KEY in n.meta[CUSTOM_KEY]
+            ):
+                ref_handle_to_nodes[n.meta[CUSTOM_KEY][NUMERIC_DEBUG_HANDLE_KEY]].append(n)
+
+        # If you re-enable XNNPackQuantizer, the debug handle 3 will be removed from weights.
+        # quantizer = XNNPACKQuantizer().set_global(
+        #     get_symmetric_quantization_config(is_per_channel=False)
+        # )
+        # m = prepare_pt2e(m, quantizer)
+        # m(*example_inputs)
+        # m = convert_pt2e(m)
+        # Re-exporting should not affect the debug handles.
+        quant_exported = export_trace._export(m, example_inputs, pre_dispatch=True)
+        quant_handle_to_nodes = defaultdict(list)
+        for n in quant_exported.graph.nodes:
+            if (
+                CUSTOM_KEY in n.meta
+                and NUMERIC_DEBUG_HANDLE_KEY in n.meta[CUSTOM_KEY]
+            ):
+                quant_handle_to_nodes[n.meta[CUSTOM_KEY][NUMERIC_DEBUG_HANDLE_KEY]].append(n)
+        # This is the bug, bias and conv have the same debug handle.
+        self.assertEqual(quant_handle_to_nodes[3][0].name, "p_getattr_l__self___conv___0___weight")
+        self.assertEqual(quant_handle_to_nodes[3][1].name, "p_getattr_l__self___conv___0___bias")
+        self.assertEqual(quant_handle_to_nodes[3][2].name, "conv2d")
```

Comment:
The bug is here, it should only have a single node for handle 3, and that should be conv2d. The weight and bias shouldn't have a handle at all (they're placeholders)